### ISSUE TITLE
Defaults: email_from/email_return_path `sender` still best?

### ISSUE THREAD

**Issue by peter-** [filesender/filesender#1820]
Defaulting to `sender` cannot work in deployments allowing logins from multiple domains:

https://github.com/filesender/filesender/blob/d4bda8035c7553b66caa9c63a49a22d50d55fb6d/includes/ConfigDefaults.php#L177-L178

Unless, of course, any and all end user email adresses recieved by FileSender during login (to be set as `sender` values) are limited to DNS domain parts that explicitly allow the MTA running on or used by that FileSender machine to send emails from that domain by
* either supplying a DKIM key pair to use (for signing of any such and only the appropriate emails)
* or by allow-listing the IP address/es used by (the MTA providing outgoing mail services for) FileSender via SPF in their DNS records
* or both.

In other words, the current default only works if you make sure that the only email addresses FileSender recieves during login are from the same DNS domain FileSender runs in.  
Or if you're prepared to do a potentitally significant amount of non-FileSender-related work to ensure such emails are actually deliverable these days.

**Comment by WebSpider** [comment]
Actually, for the installs I'm involved in, this is set to a static no-reply address these days, for reasons Peter lays out.

**Comment by monkeyiq** [comment]
Perhaps we should have a new configuration variable for the sites static no-reply address. This might even default to something like `noreply@{domain from site_url}` or just `email_reply_to`.

and change these `email_from` and `email_return_path` defaults to use that address instead?

**Comment by peter-** [comment]
No new configuration variable needed, people can just supply the email address as value of the aforementioned, existing variables. Like @WebSpider said they do.

My question was whether `sender` was still a sane default, implying that it was not, with the exceptions spelled out above.

In other words: Should defaults only work for very limited deployment situations (essentially internal to a single institution or contracted out with additional DKIM/SPF configuration to allow spoofing of the users' email addresses) or should they be suitable for federated deployments as well?  
The answer to that question will tell you whether those defaults should be kept or changed.

**Comment by meijer** [comment]
Board decision made. changing defaults sounds like a good idea, we suggest to leave it to the two of you whether to implement @WebSpider 's suggestion or go for something else. Just if you're going for something else, "[no-reply@no-domain.dom](mailto:no-reply@no-domain.dom)" might have future consequences ;)

**Comment by WebSpider** [comment]
> Perhaps we should have a new configuration variable for the sites static no-reply address. This might even default to something like `noreply@{domain from site_url}` or just `email_reply_to`.
> 
> and change these `email_from` and `email_return_path` defaults to use that address instead?

I think changing the default for all these makes sense. Would it be sufficient to just use `site_url` here as a basis to get the hostname using php's `parse_url` ?

**Comment by WebSpider** [comment]
This has been discussed and a fix for this has been merged. If you feel this is still an issue, feel free to reopen!
